coriolis/tests/scheduler/rpc/test_server.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2023 Cloudbase Solutions Srl
+# All Rights Reserved.
+
+from unittest import mock
+
+import ddt
+
+
+from coriolis import exception
+from coriolis.db import api as db_api
+from coriolis.tests import test_base, testutils
+from coriolis.scheduler.rpc import server
+
+
+@ddt.ddt
+class SchedulerServerEndpointTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Scheduler Worker RPC server."""
+
+    def setUp(self):
+        super(SchedulerServerEndpointTestCase, self).setUp()
+        self.server = server.SchedulerServerEndpoint()
+
+    @mock.patch.object(db_api, "get_regions")
+    @mock.patch.object(db_api, "get_services")
```

Comment:
I don't feel like mocking these here makes sense in the unit test scenario. I get what you tried doing here, but it'd make more sense to test `get_workers_for_specs` as a "unit", so please mock `_get_all_worker_services` instead, and move the `get_services` mock in future a `test_get_all_worker_services` case. In this particular case you should really just check `_get_all_worker_services` is called. Apply the same for the rest of the methods (mock `_filter_regions`, etc.)

Reply:
Makes sense. Fixed. Though this means that some of the exceptions in the config yaml need to be tested in different unit tests when testing the relevant functions, so those test configurations are removed for now.